### PR TITLE
Added task telemetry

### PR DESCRIPTION
**Pull request by chshrikh**

### CODE REVIEW

File: Tasks/AzureFileCopy/Utility.ps1 [hunk 1]
```diff
@@ -159,7 +127,7 @@ function Validate-AzurePowershellVersion
 
     if(!$versionCompatible)
     {
-        Write-TaskSpecificTelemetry "PREREQ_UnsupportedAzurePSVersion"
+        Write-Telemetry "Task_InternalError" "UnsupportedAzurePSVersion"
```
Comment (kmkumaran) [inline Tasks/AzureFileCopy/Utility.ps1]: I assume internal errors are also bad and it is actionable by us. Correct? If so, do we have enough info logged to take corrective action.

File: Tasks/PowerShellOnTargetMachines/PowerShellOnTargetMachines.ps1 [hunk 2]
```diff
@@ -63,27 +64,11 @@ if ($enableDetailedLoggingString -ne "true")
...
-  "PREREQ_StorageAccountNotFound" = "PREREQ_StorageAccountNotFound";
-  "AZUREPLATFORM_UnknownGetRMVMError" = "AZUREPLATFORM_UnknownGetRMVMError";
-  "DEPLOYMENT_FetchPropertyFromMap" = "DEPLOYMENT_FetchPropertyFromMap";
-  "PREREQ_UnsupportedAzurePSVersion" = "PREREQ_UnsupportedAzurePSVersion";
-  "DEPLOYMENT_CSMDeploymentFailed" = "DEPLOYMENT_CSMDeploymentFailed";
-  "PREREQ_InvalidServiceConnectionType" = "PREREQ_InvalidServiceConnectionType";
-  "PREREQ_AzureRMModuleNotFound" = "PREREQ_AzureRMModuleNotFound";
-  "PREREQ_InvalidFilePath" = "PREREQ_InvalidFilePath";
-  "DEPLOYMENT_PerformActionFailed" = "DEPLOYMENT_PerformActionFailed"
+  "Input_Validation" = "Input_Validation_Error";
```
Comment (kmkumaran) [inline Tasks/PowerShellOnTargetMachines/PowerShellOnTargetMachines.ps1]: No code duplication.

File: Tasks/PowerShellOnTargetMachines/PowerShellOnTargetMachines.ps1 [hunk 3]
```diff
@@ -94,26 +79,16 @@ function Write-Telemetry
     [string]$codeKey,
 
     [Parameter(Mandatory=$True,Position=2)]
```
Comment (kmkumaran) [inline Tasks/PowerShellOnTargetMachines/PowerShellOnTargetMachines.ps1]: Refactor to ensure all tasks use same code if required.

File: Tasks/PowerShellOnTargetMachines/PowerShellOnTargetMachines.ps1 [hunk 4]
```diff
@@ -53,7 +53,8 @@ $ErrorActionPreference = 'Stop'
 $deploymentOperation = 'Deployment'
 
 $envOperationStatus = "Passed"
-$telemetrySet = $false
+
+$taskId = "3B5693D4-5777-4FEE-862A-BD2B7A374C68"
```
Comment (kmkumaran) [inline Tasks/PowerShellOnTargetMachines/PowerShellOnTargetMachines.ps1]: Same comment. Avoid hard coding again.

File: Tasks/AzureFileCopy/Utility.ps1 [hunk 5]
```diff
@@ -2,66 +2,34 @@
...
-
-  "ENABLEWINRM_ProvisionVmCustomScriptFailed" = "ENABLEWINRM_ProvisionVmCustomScriptFailed"
-  "ENABLEWINRM_ExecutionOfVmCustomScriptFailed" = "ENABLEWINRM_ExecutionOfVmCustomScriptFailed"
-  "ADDWINRM_NetworkSecurityRuleConfigFailed" = "ADDWINRM_NetworkSecurityRuleConfigFailed"
-
-  "FILTERING_IncorrectFormat" = "FILTERING_IncorrectFormat";
-  "FILTERING_NoVMResources" = "FILTERING_NoVMResources";
-  "FILTERING_MachinesNotPresentInRG" = "FILTERING_MachinesNotPresentInRG"
+  "Input_Validation" = "Input_Validation_Error";
+  "Task_InternalError" = "Task_Run_Time_Error";
```
Comment (kmkumaran) [inline Tasks/AzureFileCopy/Utility.ps1]: Why Internal error in key and Run time error in value?

File: Tasks/AzureFileCopy/Utility.ps1 [hunk 6]
```diff
@@ -2,66 +2,34 @@
...
+  "DTLSDK_Error" = "DTL_SDK_ERROR";
```
Comment (chshrikh) [inline Tasks/AzureFileCopy/Utility.ps1]: sure

File: Tasks/AzureFileCopy/Utility.ps1 (hunk 1 above)
Comment (chshrikh) [inline Tasks/AzureFileCopy/Utility.ps1]: yes, as in this case we are saying "UnsupportedAzurePSVersion" some other like - ClassicStorageAccountNotFound, StorageAccountNotFound

**Pull request merged**